types/index.d.ts

```diff
@@ -343,7 +343,13 @@ declare namespace Realm {
         code: number;
     }
 
-    type ErrorCallback = (session: App.Sync.Session, error: SyncError) => void;
+    interface ClientResetError {
+        name: string;
```

Comment:
I think you want `name: "ClientReset";` here. Ideally you could make 339 be something like `name: string - "ClientReset";` so that it would know that that type is everything else, but I don't see a way to express that in the TS type system. @kraenhansen any ideas?

Also `readOnly: true`;